### PR TITLE
Check if the recursor is not auth for zone in a RR

### LINKED ISSUE #2600: Recursor 3.7.3 : Locally defined auth zones not overriding external answers
I have the following zones and records definition on an authoritative server (running powerdns ;) ) :

- global.isp.com :
pfs.global.isp.com IN A 10.0.0.1
- box.isp.com : 
service.box.isp.com IN CNAME pfs.global.isp.com

=> by default querying a recursor for "service.box.isp.com" returns IP address 10.0.0.1

Now I run several pdns recursor servers for specific customers where I need to "redefine" the IP address of pfs.global.isp.com :
I use "auth-zones=global.isp.com=myfile" setting of pdns recursor where "myfile" contains (besides the regular SOA fields) :
pfs.global.isp.com IN A 10.1.1.1

But when querying the specific resolvers for "service.box.isp.com", I get as answer "10.0.0.1"
=> the resolver retrieves from the authoritative server an answer containing both service.box.isp.com and the pfs.global.isp.com record pointing to IP 10.0.0.1 (as both zones are hosted on the same authoritative server) and does not take into account the local definition of global.isp.com zone

When dumping the cache after initial startup, we see that auth zones records are not yet in the cache (shouldn't be put in the cache since the start-up ?).
When querying for "pfs.global.isp.com" on the specific resolvers, I get the correct answer 10.1.1.1 and the record appears in the cache. Trace shows an explicit  "checking auth storage" for that query.
When querying for "service.box.isp.com" on the specific resolvers, the "official" record from the authoritative server is also put in the cache that now contains both definitions of the record ... but only the last one (10.0.0.1) is used from the cache to answer to the query

This behavior exists at least since version 3.4.x

### PR DESCRIPTION
**Pull request by pieterlexis**
This patch checks every resource record name in a packet from an
authoritative server and determines if we (the recursor) are authoritative
for a zone where the qname is part of. If this is the case, we don't
allow the use of that record.

Closes #2600 

ToDo:
- [x] Tests
- [ ] Review

### CODE REVIEW

**Comment by Habbie** [comment]
Could there be a similar issue with forward-zones?

**Comment by pieterlexis** [comment]
possibly, but these zones are in the same `t_sstorage->domainmap` (if I read `reczones.cc` correctly) so this patch should fix all these cases.

**Comment by pieterlexis** [comment]
Ready for review!

File: pdns/syncres.cc [hunk 1]
```diff
@@ -3,7 +3,7 @@
     Copyright (C) 2003 - 2014  PowerDNS.COM BV
 
     This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License version 2 as published
```
Comment (cmouse) [inline pdns/syncres.cc]: whitespace

File: pdns/syncres.cc [hunk 2]
```diff
@@ -91,10 +91,10 @@ SyncRes::SyncRes(const struct timeval& now) :  d_outqueries(0), d_tcpoutqueries(
 int SyncRes::beginResolve(const string &qname, const QType &qtype, uint16_t qclass, vector<DNSResourceRecord>&ret)
 {
   s_queries++;
-  
-  if( (qtype.getCode() == QType::AXFR)) 
+
+  if( (qtype.getCode() == QType::AXFR))
     return -1;
-  
+
```
Comment (cmouse) [inline pdns/syncres.cc]: unrelated changes

**Comment by pieterlexis** [comment]
again, ready for review (all whitespace fixes are in separate commits now)
